datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -1056,6 +1071,94 @@ private void onDatastreamChange(List<DatastreamGroup> datastreamGroups) {
     }
   }
 
+  /**
+   * move the partitions based on targetAssignmentInfo stored in the Zookeeper
+   */
+  private void performPartitionMovement() {
+    boolean shouldRetry = true;
+    Map<String, Set<DatastreamTask>> previousAssignmentByInstance = _adapter.getAllAssignedDatastreamTasks();
+    Map<String, List<DatastreamTask>> newAssignmentsByInstance = new HashMap<>();
+
+    try {
+      Map<String, Set<DatastreamTask>> assignmentByInstance = new HashMap<>(previousAssignmentByInstance);
+      List<DatastreamGroup> toCleanup = new ArrayList<>();
+
+
+      for (String connectorType : _connectors.keySet()) {
+        AssignmentStrategy strategy = _connectors.get(connectorType).getAssignmentStrategy();
+        Connector connectorInstance = _connectors.get(connectorType).getConnector().getConnectorInstance();
+
+        // Get the partition assignment information
+        Map<String, Optional<DatastreamGroupPartitionsMetadata>> datastreamPartitions =
+            connectorInstance.getDatastreamPartitions();
+
+        // Get the datastream Group name which have the target assignment
+        List<String> toMoveDatastream = _adapter.getDatastreamsNeedPartitionMovement(connectorType);
+
+        // Fetch all live datastreamGroups
+        List<DatastreamGroup> liveDatastreamGroups =
+            fetchDatastreamGroups().stream().filter(group1 -> connectorType.equals(group1.getConnectorName())).collect(
+                Collectors.toList());
+
+        // clean up the datastreams if they are not in live datastream
+        toMoveDatastream.stream().filter(dgName -> !liveDatastreamGroups.stream().map(DatastreamGroup::getName)
+            .collect(Collectors.toList()).contains(dgName)).forEach(obsoleteDs -> _adapter.cleanUpPartitionMovement(connectorType, obsoleteDs));
+
+        // Filtered all live datastreamGrou as we process only datastream which have
+        // both partition assignment info and the target assignment
+        List<DatastreamGroup> toProcessedDatastreamGroups =
+            liveDatastreamGroups.stream().filter(group2 -> toMoveDatastream.contains(group2.getName()))
+                .filter(group3 -> datastreamPartitions.keySet().contains(group3.getName()))
+                .collect(Collectors.toList());
+
+        for (DatastreamGroup dg : toProcessedDatastreamGroups) {
+          DatastreamGroupPartitionsMetadata subscribedPartitions = connectorInstance.getDatastreamPartitions().get(dg.getName())
```

Comment:
I'm inclined to keep the current behavior. It might be easier to debug to see all or none behavior especially the currentAssignment store things from every datastream group.